src/frontend/src/optimizer/plan_node/batch_hop_window.rs

```diff
@@ -67,24 +67,6 @@ impl ToDistributedBatch for BatchHopWindow {
         let new_input = self.input().to_distributed()?;
         Ok(self.clone_with_input(new_input).into())
     }
-
-    fn to_distributed_with_required(
-        &self,
-        required_order: &Order,
-        required_dist: &RequiredDist,
-    ) -> Result<PlanRef> {
-        let input_required = self
-            .logical
-            .o2i_col_mapping()
-            .rewrite_required_distribution(required_dist);
-        let new_input = self
-            .input()
-            .to_distributed_with_required(required_order, &input_required)?;
-        let new_logical = self.logical.clone_with_input(new_input);
-        let batch_plan = BatchHopWindow::new(new_logical);
-        let batch_plan = required_order.enforce_if_not_satisfies(batch_plan.into())?;
-        required_dist.enforce_if_not_satisfies(batch_plan, required_order)
-    }
```

Comment:
This causes reduntant exchange. See plan diff

Reply:
the hop operator will generate a multiplication of its input rows, so shuffling its input instead of its output will reduce the shuffling data communication.